Mapbox/MapboxMapsStyle/Types/Color.swift

```diff
@@ -1,52 +1,49 @@
 import UIKit
 
-public typealias ColorRepresentable = String
-extension ColorRepresentable {
+/// Container to represent `UIColor`for use by the map renderer
+public struct ColorRepresentable: Codable, Equatable {
+
+    /// String representation of a `UIColor` used by the renderer
+    public let colorRepresentation: String?
 
     /// Create a string representation of a `UIColor`
+    /// - Parameter color: A `UIColor` instance in the sRGB color space
+    /// - Returns: Initializes a `ColorRepresentable` instance if the `color` is in sRGB color space.
     public init(color: UIColor) {
         var red: CGFloat = 0.0
         var green: CGFloat = 0.0
         var blue: CGFloat = 0.0
         var alpha: CGFloat = 0.0
-        color.getRed(&red, green: &green, blue: &blue, alpha: &alpha)
-        self = "rgba(\(red * 255.0), \(green * 255.0), \(blue * 255.0), \(alpha))"
+        let success = color.getRed(&red, green: &green, blue: &blue, alpha: &alpha)
+        let validColorComponents = Self.isValidColor(red: red, green: red, blue: blue, alpha: alpha)
+        if success && validColorComponents {
+            self.colorRepresentation = "rgba(\(red * 255.0), \(green * 255.0), \(blue * 255.0), \(alpha))"
+        } else {
+            fatalError("Please use a color in the sRGB color space")
+        }
     }
-}
-
-public extension UIColor {
-
-    /// Initialize a `UIColor` from a `ColorRepresentable`
-    convenience init?(hex: ColorRepresentable?) {
-        guard let hex = hex else { return nil }
-
-        let red, green, blue, alpha: CGFloat
 
-        if hex.hasPrefix("#") {
-            let start = hex.index(hex.startIndex, offsetBy: 1)
-            let hexColor = String(hex[start...])
-
-            if hexColor.count == 8 {
-                let scanner = Scanner(string: hexColor)
-                var hexNumber: UInt64 = 0
-
-                if scanner.scanHexInt64(&hexNumber) {
-                    red = CGFloat((hexNumber & 0xff000000) >> 24) / 255
-                    green = CGFloat((hexNumber & 0x00ff0000) >> 16) / 255
-                    blue = CGFloat((hexNumber & 0x0000ff00) >> 8) / 255
-                    alpha = CGFloat(hexNumber & 0x000000ff) / 255
+    /// Checks if all color components are within the 0-1 range
+    internal static func isValidColor(red: CGFloat, green: CGFloat, blue: CGFloat, alpha: CGFloat) -> Bool {
+        let validRange = 0.0...1.0
+        return validRange ~= Double(red)
+            && validRange ~= Double(green)
+            && validRange ~= Double(blue)
+            && validRange ~= Double(alpha)
```

Comment:
```suggestion
        let validRange: CGFloat = 0.0...1.0
        return validRange ~= red
            && validRange ~= green
            && validRange ~= blue
            && validRange ~= Double(alpha)
```

Does that work?

Reply:
I'd tried that, doesn't work unfortunately.

Reply:
What about this:

```suggestion
        let validRange: ClosedRange<CGFloat> = 0.0...1.0
        return validRange ~= red
            && validRange ~= green
            && validRange ~= blue
            && validRange ~= alpha
```